src/clincoded/static/components/hpo_term_modal.js

```diff
@@ -0,0 +1,193 @@
+'use strict';
+import React from 'react';
+import PropTypes from 'prop-types';
+import createReactClass from 'create-react-class';
+import _ from 'underscore';
+import * as curator from './curator';
+import { external_url_map } from './globals';
+import { RestMixin } from './rest';
+import { FormMixin, Input } from '../libs/bootstrap/form';
+import ModalComponent from '../libs/bootstrap/modal';
+
+var HpoTermModal = createReactClass({
+    mixins: [RestMixin, FormMixin],
+
+    propTypes: {
+        addHpoTermButton: PropTypes.object.isRequired, // Button to trigger HPO modal
+        inElim: PropTypes.bool, // Used to differentiate between normal and "in elim" modal
+        savedHpo: PropTypes.array, // HPO saved from parent
+        savedElimHpo: PropTypes.array, // HPO in elimination saved from parent
+        passHpoToParent: PropTypes.func, // Function used to pass HPO data to respective parent
+        passElimHpoToParent: PropTypes.func // Function used to pass Elim HPO data to parent
+    },
+
+    getInitialState: function() {
+        return {
+            hpoWithTerms: [],
+            hpoElimWithTerms: []
+        };
+    },
+
+    handleModalClose() {
+        this.child.closeModal();
+    },
+
+    componentDidMount() {
+        const savedHpo = this.props.savedHpo ? this.props.savedHpo : null;
+        const savedElimHpo = this.props.savedElimHpo ? this.props.savedElimHpo : null;
+        if (savedHpo && savedHpo.length) {
+            this.setState({ hpoWithTerms: savedHpo });
+        }
+        if (savedElimHpo && savedElimHpo.length) {
+            this.setState({ hpoElimWithTerms: savedElimHpo });
+        }
+    },
+
+    saveForm(e) {
+        e.preventDefault(); e.stopPropagation();
+        let hpoWithTermsList = this.state.hpoWithTerms;
+        let hpoElimWithTermsList = this.state.hpoElimWithTerms;
```

Comment:
Suggestion: change to following so duplicated hpo will be removed.
        let hpoWithTermsList = _.uniq(this.state.hpoWithTerms);
        let hpoElimWithTermsList = _.uniq(this.state.hpoElimWithTerms);
        this.setState({ hpoWithTerms: hpoWithTermsList });
        this.setState({ hpoElimWithTerms: hpoElimWithTermsList });